server/src/main/java/org/opensearch/cluster/routing/allocation/decider/AllocationDeciders.java

```diff
@@ -236,6 +238,72 @@ public Decision canForceAllocatePrimary(ShardRouting shardRouting, RoutingNode n
         return ret;
     }
 
+    @Override
+    public Decision canAllocateAnyShardToNode(RoutingNode node, RoutingAllocation allocation) {
+        Decision.Multi ret = new Decision.Multi();
+        for (AllocationDecider decider : allocations) {
+            Decision decision = decider.canAllocateAnyShardToNode(node, allocation);
```

Comment:
Decider invocation here is the only difference in the logic to determine ``canAllocateAnyShardToNode ``, ``canMoveAway`` and ``canMoveAnyShard``. Shall we move the logic to a common method and provide this as a Supplier instead.

Reply:
The current paradigm is inline with rest of the methods in this class. There was small else block improvement that I have done.
I spent sometime on abstracting all methods in class but realized that logic is fairly simple and better read inline